app/services/namespaces/group_unshare_service.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Namespaces
+  # Service used to Destroy NamespaceGroupLinks
+  class GroupUnshareService < BaseService
+    NamespaceGroupUnshareError = Class.new(StandardError)
+    attr_accessor :group_id, :namespace, :max_group_access_role
+
+    def initialize(user, group_id, namespace)
+      super(user, namespace)
+      @group_id = group_id
+      @namespace = namespace
+    end
+
+    def execute
+      authorize! namespace, to: :unshare_namespace_with_group?
+
+      namespace_group_link = NamespaceGroupLink.find_by(group_id:, namespace:)
+
+      raise NamespaceGroupUnshareError, 'Namespace to group link does not exist' if namespace_group_link.nil?
```

Comment:
**nitpick (blocking):** This message should be translated as you are adding it to the errors below.

Reply:
Added translation in 66aaa75edf6c80d24a88cb6abfc8a07a4e4d90a6